File: jscomp/runtime/caml_js_exceptions.ml


type  t = 
  | Any : 'a ->  t [@@unboxed]

exception Error of  t 


(**   
   This function has to be in this module Since 
   [Error] is defined here 
*)
let internalToOCamlException (e : Obj.t) =
  if Caml_exceptions.caml_is_extension e  then
    (Obj.magic e  : exn)
  else Error (Any e)

let caml_as_js_exn exn =   
  match exn with 
  | Error t ->  
    Some t
  | _ -> None 

File: lib/es6/caml_js_exceptions.js


import * as Caml_exceptions from "./caml_exceptions.js";

function internalToOCamlException(e) {
  if (Caml_exceptions.is_extension(e)) {
    return e;
  } else {
    return {
      RE_EXN_ID: "JsError",
      _1: e
    };
  }
}

let Obj;

let $$Error = "JsError";

export {
  Obj,
  $$Error,
  internalToOCamlException,
}
/* No side effect */
